sidecar/internal/service/reclaimspace.go

```diff
@@ -88,14 +130,25 @@ func (rs *ReclaimSpaceServer) NodeReclaimSpace(
 	pvName := req.GetPvName()
 	klog.Info(pvName)
 
-	// TODO: get following params using kubeClient and pvName
+	pv, err := rs.kubeClient.CoreV1().PersistentVolumes().Get(context.TODO(), pvName, metav1.GetOptions{})
+	volID := pv.Spec.PersistentVolumeSource.CSI.VolumeHandle
+
+	deviceSecret, err := getSecret(rs.kubeClient,
+		pv.Spec.CSI.NodeStageSecretRef.Namespace,
```

Comment:
same here, check for the existence of the secret options in the PV before using them